java/src/main/java/com/saucelabs/saucebindings/configs/SauceConfigs.java

```diff
@@ -0,0 +1,61 @@
+package com.saucelabs.saucebindings.configs;
+
+import com.saucelabs.saucebindings.JobVisibility;
+import com.saucelabs.saucebindings.SauceOptions;
+import com.saucelabs.saucebindings.SaucePlatform;
+
+import java.util.List;
+import java.util.Map;
+
+public class SauceConfigs<T extends SauceConfigs<T>> {
+    // Needs to be instantiated in subclass
+    SauceOptions sauceOptions = null;
+
+    // Available on all configs desktop & mobile
+
+    // Override this in subclasses to ensure valid enum
+    public T setPlatformName(SaucePlatform platform) {
+        sauceOptions.setPlatformName(platform);
+        return (T) this;
+    }
+
+    public T setName(String name) {
+        sauceOptions.setName(name);
+        return (T) this;
+    }
+
+    public T setBuild(String build) {
+        sauceOptions.setBuild(build);
+        return (T) this;
+    }
+    
+    public T setTags(List<String> tags) {
+        sauceOptions.setTags(tags);
+        return (T) this;
+    }
+    
+    public T setCustomData(Map<String, Object> data) {
+        sauceOptions.setCustomData(data);
+        return (T) this;
+    }
+    
+    // This corresponds to "public" keyword
+    public T setJobVisibility(JobVisibility visibility) {
+        sauceOptions.setJobVisibility(visibility);
+        return (T) this;
+    }
+    
+    public T setTunnelIdentifier(String identifier) {
+        sauceOptions.setTunnelIdentifier(identifier);
+        return (T) this;
+    }
+
+    public T setParentTunnel(String identifier) {
+        sauceOptions.setParentTunnel(identifier);
+        return (T) this;
+    }
+
+    public SauceOptions build() {
```

Comment:
It feels you'd like to follow the same structure of browser options in Selenium (where `EdgeOptions` extends `ChromiumOptions`, etc...), so maybe you can get rid of this `build` method and only pass instanciated objects?

Reply:
The goal is to only show users allowed capabilities. So SauceOptions allows everything, and the config class is restricted to only applicable values, but rather than a bunch of other constructors, we just convert the configs class into a sauce options and have that get passed into the session.